mapbox/libjava-services/src/main/java/com/mapbox/services/api/directions/v5/MapboxDirections.java

```diff
@@ -578,5 +575,25 @@ public MapboxDirections build() throws ServicesException {
       }
       return new MapboxDirections(this);
     }
+
```

Comment:
On a second thought, I think we should fix this in a simpler way.
- `setOrigin(Position origin)` should set a private `Position origin` value.
- `setDestination(Position destination)` should set a private `Position destination` value.
- `setCoordinates(List<Position> coordinates)` should set a private `List<Position> coordinates` value.

Then, at build time we check if both (`origin`, `destination`) and  `coordinates` have been set. If the developer has mixed them we through a `ServicesException`.